### PR TITLE
Overlapping query fix

### PR DESCRIPTION
**Pull request by AronBuzogany**
This Pull Request fixes issue #4 by using web worker for processing the query and terminating it whenever another query is selected

### CODE REVIEW

File: src/right-field/results-table/ResultsTable.jsx [hunk 1]
```diff
@@ -16,45 +16,52 @@ if(config.queryFolder.substring(config.queryFolder.length-1) !== '/'){
...
       }
       
       return [...old, newValues]}
       
     )
+    
+    useEffect(() => {configureQueryWorker(adder, setVariables, setQuerying)}, [])
+    
 
     let onqueryChanged = () => {
```
Comment (pheyvaer) [inline src/right-field/results-table/ResultsTable.jsx:55]: Why is this variable declared with `let`?

File: src/right-field/results-table/ResultsTable.jsx [hunk 2]
```diff
@@ -100,77 +131,19 @@ function generateColumn(variable, size){
  * @param {Function} adder a function which handles what happens with every variable result
```
Comment (pheyvaer) [inline src/right-field/results-table/ResultsTable.jsx:131]: This parameter is not present in the signature of the function.

**Review by pheyvaer — CHANGES_REQUESTED** [review]
Besides the specific comments, make sure that overall the same indentation is used. Preferably 2 spaces.

File: src/workers/worker.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      postMessage({type: 'metadata', metadata: {variables: metadata.variables, queryType: execution.resultType}})
+
+      queryTypeHandlers[execution.resultType](await execution.execute())
+
+    }
+    catch(error){
+      handleBindingStreamFail(error)
+    }
+     
+  }
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/workers/worker.js]: I have completely reformated the file in [commit 6914cc9](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/6914cc93cb16d013f2a2e8b3578e1b0d8521adbe)

File: src/workers/worker.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+function configureBool(result){
+  postMessage({type: 'result', result: result})
+  postMessage({type: "end", message: "blank"})
+}
+
+function configureStream(stream, dataParser = (data) => {return JSON.stringify(data)}){
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/workers/worker.js]: I have added and adjusted documentation in [commit dc42dd6](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/dc42dd65bff6c54ddc6d7df1ff67c6529ca74e48)

File: src/workers/worker.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  stream.on('error', (e) => {
+      postMessage({type: "error", error: e})
+  })
+
+  stream.on('end', () => {
+      postMessage({type: "end", message: "blank"})
+  })
+}
+
+function configureQuadStream(quadStream){
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/workers/worker.js]: I have added and adjusted documentation in [commit dc42dd6](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/dc42dd65bff6c54ddc6d7df1ff67c6529ca74e48)

File: src/workers/worker.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    configureStream(quadStream)
+}
+
+function configureBindingStream(bindingStream){
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/workers/worker.js]: I have added and adjusted documentation in [commit dc42dd6](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/dc42dd65bff6c54ddc6d7df1ff67c6529ca74e48)

File: src/right-field/results-table/ResultsTable.jsx [hunk 7]
```diff
@@ -16,45 +16,52 @@ if(config.queryFolder.substring(config.queryFolder.length-1) !== '/'){
...
-const QueryEngine = require('@comunica/query-sparql').QueryEngine;
-const myEngine = new QueryEngine()
-
+let queryWorker = undefined
 
 /**
  * 
- * @param {query} props.query The query (as defined in the config file) that should be executed and results displayed in the table. 
+ * @param {query} props.query The query undefined(as defined in the config file) that should be executed and results displayed in the table. 
+ * @param {query} props.query The query undefined(as defined in the config file) that should be executed and results displayed in the table.
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/right-field/results-table/ResultsTable.jsx]: I have added and adjusted documentation in [commit dc42dd6](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/dc42dd65bff6c54ddc6d7df1ff67c6529ca74e48)

File: src/right-field/results-table/ResultsTable.jsx [hunk 8]
```diff
@@ -83,6 +90,30 @@ function ResultsTable(props){
...
+        variablesMain = data.metadata.variables.map(val => val.value)
+        variableSetter(variablesMain)
+        break; 
+      }
+    }
+  }
+}
+
+
 function generateColumn(variable, size){
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/right-field/results-table/ResultsTable.jsx]: I have added and adjusted documentation in [commit dc42dd6](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/dc42dd65bff6c54ddc6d7df1ff67c6529ca74e48)

File: src/right-field/results-table/ResultsTable.jsx (hunk 2 above)
Comment (AronBuzogany) [reply src/right-field/results-table/ResultsTable.jsx]: I have removed this line in [commit dc42dd6](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/dc42dd65bff6c54ddc6d7df1ff67c6529ca74e48)

File: src/right-field/results-table/ResultsTable.jsx (hunk 1 above)
Comment (AronBuzogany) [reply src/right-field/results-table/ResultsTable.jsx]: Probably something I forget to change, this is now adjusted respectively in [commit 4b42c9a](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/4b42c9a4db57771ec72ec43023cb3daac5879280)

File: src/right-field/results-table/ResultsTable.jsx [hunk 9]
```diff
@@ -4,180 +4,170 @@ import config from "../../config.json"
...
+ * @param {Function} adder function which processes the result, takes the result as argument
+ * @param {Function} variableSetter setter function for the variables, takes a list of variable names
+ * @param {Function} setIsQuerying boolean setter function for whether the worker is still executing a query or not
+ */
+function configureQueryWorker(adder, variableSetter, setIsQuerying) {
+  queryWorker = new QueryWorker()
+  let variablesMain = []
+  queryWorker.onmessage = ({ data }) => {
+    switch (data.type) {
+      case 'result':
```
Comment (pheyvaer) [inline src/right-field/results-table/ResultsTable.jsx:113]: Either use single quotes or double quotes, but not both.

**Review by pheyvaer — CHANGES_REQUESTED** [review]
Also fix the following warnings that `npm start` shows
```shell
WARNING in [eslint] 
src/right-field/results-table/ResultsTable.jsx
  Line 5:16:   '_' is defined but never used                                                                                      no-unused-vars
  Line 37:79:  React Hook useEffect has a missing dependency: 'adder'. Either include it or remove the dependency array           react-hooks/exhaustive-deps
  Line 60:6:   React Hook useEffect has a missing dependency: 'onQueryChanged'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  Line 112:5:  Expected a default case                                                                                            default-case

src/typeMapper.js
  Line 4:23:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images  jsx-a11y/alt-text
```

File: src/right-field/results-table/ResultsTable.jsx (hunk 9 above)
Comment (AronBuzogany) [reply src/right-field/results-table/ResultsTable.jsx]: I have adjusted this in [commit a9ae377](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/a9ae3773cabbc1aed0d43c54d72e5493048f4625)

File: src/workers/worker.js [hunk 10]
```diff
@@ -0,0 +1,109 @@
+import { QueryEngine } from "@comunica/query-sparql"
+
+
+let myEngine = new QueryEngine()
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/workers/worker.js]: It indeed can be, I have adjusted this in [commit b5b2517](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/b5b2517e34792e31a0b20cd78a51ab4e0e8c6582)

File: src/workers/worker.js [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+onmessage = (selectedQuery) => {
+  executeQuery(selectedQuery.data.selectedQuery)
+}
+
+/**
+ * A function that executes a given query and processes every result as a stream based on the functions provided.
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/workers/worker.js:11]: This description has been changed in [commit dc42dd6](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/dc42dd65bff6c54ddc6d7df1ff67c6529ca74e48)

File: src/right-field/results-table/ResultsTable.jsx [hunk 12]
```diff
@@ -4,180 +4,170 @@ import config from "../../config.json"
...
+      }
     }
   }
-  return list
 }
 
 /**
- * Handles the event whenever an error occurs during query execution. 
- * @param {Error} error object returned by the communica engine whenever a problem occurs during query execution.
+ *
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/right-field/results-table/ResultsTable.jsx]: I have provided extra descriptons in [commit a56fcef](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/a56fcef2c5a6c5b8ccd9259447b3171661297632)

File: src/right-field/results-table/ResultsTable.jsx [hunk 13]
```diff
@@ -4,180 +4,170 @@ import config from "../../config.json"
...
+
+  setter(old => { return [...old, newValues] })
 }
 
-function extendList(list, newList){
-  for(let value of newList){
-    if(!list.includes(value)){
-      list.push(value)
+/**
+ *
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/right-field/results-table/ResultsTable.jsx]: I have provided extra descriptons in [commit a56fcef](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/a56fcef2c5a6c5b8ccd9259447b3171661297632)

File: src/right-field/results-table/ResultsTable.jsx [hunk 14]
```diff
@@ -4,180 +4,170 @@ import config from "../../config.json"
...
   config.queryFolder = `${config.queryFolder}/`
 }
 
-const QueryEngine = require('@comunica/query-sparql').QueryEngine;
-const myEngine = new QueryEngine()
-
+let queryWorker = undefined
 
 /**
  *
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/right-field/results-table/ResultsTable.jsx:22]: I have provided extra descriptons in [commit a56fcef](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/a56fcef2c5a6c5b8ccd9259447b3171661297632)

**Review by pheyvaer — APPROVED** [review]

**Pull request merged**